### PR TITLE
Fixes #7

### LINKED ISSUE #7: Automate file reading to populate json object
* From reading in file (fcsv, csv) to populating json object
* fid name, x, y, z coordinates,

### PR DESCRIPTION
**Pull request by kaitj**
- Added functionality for fcsv/csv parsing into json object 
- Adds requirements.txt file
- Add a .gitignore file (from local development)

@tkkuehn - Should be able to use additions for #8

### CODE REVIEW

File: var/www/fidvalidator_pythonanywhere_com_wsgi.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import sys
-path = '/home/fidvalidator/'
+path = '/home/fidvalidator'
```
[the remark this answers is not in the document]
Comment (kaitj) [reply var/www/fidvalidator_pythonanywhere_com_wsgi.py]: Was an accidently local push, but if PythonAnywhere generates automatically, can remove all together.

File: home/fidvalidator/model_auto.py [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+
+
+def csvtojson(in_file):
+    """ Parse .fscv / .csv files and write to json object """
+    # Irrelevant fields set to None
+    fields = (None, 'x', 'y', 'z', None, None, None, None, None,
+              None, None, None, 'desc', None)
+
+    # Read CSV
+    csvFile = open(in_file, 'r')
```
[the remark this answers is not in the document]
Comment (kaitj) [reply home/fidvalidator/model_auto.py]: Easy to change! Can definitely go back and change this if above is what we plan on moving forward with.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Click==7.0
+Flask==1.0.2
+Flask-WTF==0.14.2
```
[the remark this answers is not in the document]
Comment (kaitj) [reply requirements.txt]: Great. Let's just use the requirements.txt from #11.

**Comment by kaitj** [comment]
I believe the original version of the validator was comparing two fcsv/csv files? I agree with comments. Can't recall why we wanted to go fcsc to json. Looking back, seems redundant to go from fcsv to json for the comparison. Are there any benefits to converting to json first? 

@Park-Patrick @ostanley @jclauneuro thoughts?

To the last point of python dict and json object, Python's json library does a pretty good job of going between the two.

**Comment by jclauneuro** [comment]
+1. I don't see any advantage of creating our own internal representation in JSON, particularly given that figuring out how to represent discrete coordinates seems likely to be a priority in 2019 within the BIDS community: https://groups.google.com/forum/#!topic/bids-discussion/rQPf6DuVbh0

Nice to know that converting between object types seems relatively straightforward though. Anything else I've forgotten to comment on in this thread?

**Comment by kaitj** [comment]
Just an additional thought and maybe it was why we wanted to convert it to JSON in the first place. 

We wouldn't constrained by the limitations of a csv reader (eg. row-by-row computations) if it it is in a python dict/JSON object. I'm not aware of any way of probing the CSV directly and the limitation would be row-by-row computations. The caveat is we would have to ensure the inputted file is structured how we want it (see #8 - afids in ascending order would have to be checked before hand, or we could add that column to the JSON object for the comparison).

**Comment by jclauneuro** [comment]
Thanks for the clarification, @kaitj! That could be very well the reason why!

I have no experience with trying to parse through JSON file (for analysis, I live in an R dataframe world) but that definitely sounds like an advantage. We're ultimately only talking about 5 columns/values per "row" so the implementation cost of creating our own JSON format is likely minimal?

**Comment by ostanley** [comment]
You guys seem to have a handle on this but I agree with @kaitj that json will be more flexible long term. We could always switch back later.

**Comment by tkkuehn** [comment]
Okay, I'm on board with the JSON structure we've settled on and the current version of the PR looks good to me.

If we're going to merge this and #11, I think it makes sense to merge #11 first, bring those structural changes in here, then merge this. @kaitj if you agree I can handle that.

**Comment by ostanley** [comment]
Great then I'll update the whole thing on our python anywhere and add the links to the git repo so we are not manually uploading. Easier once these two are merged in.

**Pull request merged**